drivers/can/can_esp32_twai.c

```diff
@@ -202,9 +204,16 @@ static int can_esp32_twai_init(const struct device *dev)
 	can_esp32_twai_write_reg32(dev, TWAI_CLOCK_DIVIDER_REG, twai_config->cdr32);
 #endif /* !CONFIG_SOC_SERIES_ESP32 */
 
-	esp_intr_alloc(twai_config->irq_source, 0, can_esp32_twai_isr, (void *)dev, NULL);
+	err = esp_intr_alloc(twai_config->irq_source,
+			esp_intr_level_to_flags(twai_config->irq_priority) |
+			esp_intr_flags_check(twai_config->irq_flags) | ESP_INTR_FLAG_IRAM,
```

Comment:
Maybe a macro creating flags out of the `irq_priority` and `irq_flags` would help in usability.

Reply:
Inline functions were replaced by macros